packages/backend.ai-ui/src/components/baiClient/FileExplorer/ActionItems.tsx

```diff
@@ -0,0 +1,210 @@
+import { BAITrashBinIcon } from '../../../icons';
+import BAIFlex from '../../BAIFlex';
+import useConnectedBAIClient from '../../provider/BAIClientProvider/hooks/useConnectedBAIClient';
+import { VFolderFile } from '../../provider/BAIClientProvider/types';
+import { FolderInfoContext } from './BAIFileExplorer';
+import CreateDirectoryModal from './CreateDirectoryModal';
+import DeleteSelectedItemsModal from './DeleteSelectedItemsModal';
+import {
+  FileAddOutlined,
+  FolderAddOutlined,
+  UploadOutlined,
+} from '@ant-design/icons';
+import { useQuery } from '@tanstack/react-query';
+import { useToggle } from 'ahooks';
+import { App, Button, Dropdown, Grid, theme, Tooltip, Upload } from 'antd';
+import { RcFile } from 'antd/es/upload';
+import _ from 'lodash';
+import { use } from 'react';
+import { useTranslation } from 'react-i18next';
+import * as tus from 'tus-js-client';
+
+interface ActionItemsProps {
+  selectedItems: Array<VFolderFile>;
+  onRequestClose: (
+    success: boolean,
+    modifiedItems?: Array<VFolderFile>,
+  ) => void;
+}
+
+const ActionItems: React.FC<ActionItemsProps> = ({
+  selectedItems,
+  onRequestClose,
+}) => {
+  const baiClient = useConnectedBAIClient();
+  const { t } = useTranslation();
+  const { lg } = Grid.useBreakpoint();
+  const { token } = theme.useToken();
+  const { message } = App.useApp();
+  const { targetVFolderId, currentPath } = use(FolderInfoContext);
+  const [openCreateModal, { toggle: toggleCreateModal }] = useToggle(false);
+  const [openDeleteModal, { toggle: toggleDeleteModal }] = useToggle(false);
+
+  const { data: vfolderInfo, isFetching } = useQuery({
+    queryKey: ['vfolderInfo', targetVFolderId],
+    queryFn: () => baiClient.vfolder.info(targetVFolderId),
+    enabled: !!targetVFolderId,
+    // not using cache, always refetch
+    staleTime: 5 * 60 * 1000,
+    gcTime: 0,
+  });
+
+  const handleUpload = (fileList: Array<RcFile>) => {
+    const maxUploadSize = baiClient?._config?.maxFileUploadSize;
+    if (
+      maxUploadSize > 0 &&
+      _.some(fileList, (file) => file.size > maxUploadSize)
+    ) {
+      message.error(t('comp:FileExplorer.error.FileUploadSizeLimit'));
+      return;
+    }
+
+    // TODO: show confirmation modal before upload already existing files
+    // TODO: show progress during upload via using bai-notification
+    const uploadFiles: Array<Promise<tus.Upload>> = _.map(
```

Comment:
When uploading a folder containing many files and subfolders, the Lit version starts uploading immediately, but the React version causes the browser to freeze.